src/tools/compiletest/src/main.rs

```diff
@@ -591,6 +595,24 @@ fn collect_tests_from_dir(
     let build_dir = output_relative_path(config, relative_dir_path);
     fs::create_dir_all(&build_dir).unwrap();
 
+    if config.only_modified {
+        let Ok(Some(files)) = get_git_modified_files(Some(dir), &vec!["rs", "stderr", "fixed"]) else { return Ok(()); };
+        // Add new test cases to the list, it will be convenient in daily development.
+        let Ok(Some(untracked_files)) = get_git_untracked_files(None) else { return Ok(()); };
+        let mut all_paths = files;
+        all_paths.extend(untracked_files);
+        let mut full_paths: Vec<PathBuf> =
+            all_paths.iter().map(|f| fs::canonicalize(&f).unwrap().with_extension("rs")).collect();
+        full_paths.dedup();
```

Comment:
For this `dedup()` to be fully effective, the values need to be sorted. Are they because of the guaranteed output of the git command, or is it only an optimisation that does not affect correctness?

Reply:
emm, I just don't realize the pre-condition of `dedup`, I think we'd better sort it.

Reply:
The doc is at https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.dedup